article/article.md

```diff
@@ -0,0 +1,333 @@
+
+I'm an Android developer on a daily basis. I first came across this page among other Android documentations: [Build client-server applications with gRPC](https://developer.android.com/guide/topics/connectivity/grpc), gRPC was not completely alien to me since my coworkers had been discussing it enthusiastically for a while.
+
+gRPC is a new take on an old approach known as RPC ([Remote Procedure Call](https://en.wikipedia.org/wiki/Remote_procedure_call)), a form of inter-process communication essentially used for client-server interactions. The client can request to execute a procedure on the server as if it were a normal local procedure call thanks to the stub generated for both client and server.
+
+The typical sequence of the RPC events runs like this:
+
+- The client calls the client stub with parameters provided in the normal way
+- The client stub then builts a message (marshalling) with the parameters and makes a system call to send the message to the server 
+- The message is sent to the server by the client's local operating system 
+- The server's local operating system passes the incoming message to the server stub
+- The server stub unpacks the parameters from the message (unmarshalling)
+- At the end the server stub calls the server procedure with the parameters
+
+Base on the same principle, [gRPC](https://grpc.io/) is an open-source RPC framework initially developed by Google and currently a [Cloud Native Computing Foundation](https://www.cncf.io/) incubating project. By default, gRPC uses HTTP/2 for transport and [Protocol Buffers](https://developers.google.com/protocol-buffers) as both its Interface Definition Language (IDL) and as its underlying message interchange format.
+
+![gRPC-archi](gRPC-archi.png)
+
+# REST v.s. gRPC
+
+To grasp why gRPC is a fundemantally different approach to build web services. Let's do a [side-by-side comparison](https://speakerdeck.com/thesandlord/grpc-vs-rest-api-strat-2016) between REST and gRPC.
+
+REST is basically JSON over HTTP 1.1 and gRPC is basically Protobuf over HTTP/2 with POST and choosing gRPC for a new project is still pretty experimental at the time of writing.
+
+| JSON | Protocol Buffers |
+|------|------------------|
+| Plaintext | Binary |
+| Human Readable | Machine Readable |
+| Repetitive | Compressed |
+| Fast (De)Serialization | Faster (De)Serialization |
+| Everything supports it | Limited support |
+
+| HTTP 1.1 | HTTP/2 |
+|----------|--------|
+| Plaintext | Binary |
+| No Pipelining | Native Pipelining |
+| New Connection Per Request | Persistent TCP Connection |
+| Repetitive | Compressed |
+| Polling | Streaming |
+| Non-secure by default | TLS by default |
+| Everything supports it | Limited support |
+
+# gRPC in action: Pokédex
+
+In order to demonstrate how gRPC can be used in a real project, I decided to build a simple Pokédex, which allows me to enter a Pokémon's English name and retrieve its Pokédex number, name in French, type in French and image. 
```

Comment:
Only a suggestion
```suggestion
In order to demonstrate how gRPC can be used in a real project, I decided to build a simple Pokédex, which allows me to enter a Pokémon's English name and retrieve its Pokédex number, (French) name, (French) type and image. 
```

Reply:
For this one I'll keep the original one :)